src/main/java/fm/last/moji/impl/FileUploadOutputStream.java

```diff
@@ -80,40 +81,62 @@ public void flush() throws IOException {
   @Override
   public void close() throws IOException {
     log.debug("Close called on {}", this);
-    long size = -1L;
+
     try {
+      // Step 1. finish data writing to store node
+      long size = -1L;
       try {
         delegate.flush();
         size = delegate.getByteCount();
+        log.debug("Bytes written: {}", size);
+
+        String message = httpConnection.getResponseMessage();
+        int code = httpConnection.getResponseCode();
+        if (HttpURLConnection.HTTP_OK != code && HttpURLConnection.HTTP_CREATED != code) {
+          throw new IOException(code + " " + message);
+        }
       } finally {
         try {
           delegate.close();
-        } finally {
-          try {
-            String message = httpConnection.getResponseMessage();
-            int code = httpConnection.getResponseCode();
-            if (HttpURLConnection.HTTP_OK != code && HttpURLConnection.HTTP_CREATED != code) {
-              throw new IOException(code + " " + message);
-            } else {
-              log.debug("Status: HTTP {} - {}", code, message);
-            }
-          } finally {
-            httpConnection.disconnect();
-          }
+        } catch (Exception e) {
+          log.warn("Cannot close stream", e);
+        }
+        try {
+          httpConnection.disconnect();
+        } catch (Exception e) {
```

Comment:
Because I cannot expect what exception will be thrown by disconnect(). The intention here is just record the httpConnection cannot be closed in any case.